### ISSUE TITLE
ipcRenderer.callMain - subsequent calls return non-resolving Promises

### ISSUE THREAD

**Issue by maciejewiczow** [sindresorhus/electron-better-ipc#35]
When `ipcRenderer.callMain` gets called two or more times, the first Promise resolves normally, but subsequent calls return non-resolving Promises.
I'm using typescript and electron-webpack. I've thrown together a [quick bug repro](https://github.com/maciejewiczow/electron-better-ipc-bug-repro) repo to ilustrate the problem.
![repro](https://user-images.githubusercontent.com/37718435/85337786-639b8800-b4e1-11ea-8409-a72590b30c8b.gif)
I did a bit of investigation around the method's source code and found out that in the second call, the callbacks (onData, onError) were not called. But, when I commented out [these lines](https://github.com/sindresorhus/electron-better-ipc/blob/414a2a9db91202bb76264a0ac3a8fde8b1152568/source/renderer.js#L13-L14), the issue disappeared (at least for my test case)
```diff
ipc.callMain = (channel, data) => new Promise((resolve, reject) => {
	const {sendChannel, dataChannel, errorChannel} = util.getResponseChannels(channel);

	const cleanup = () => {
-		ipc.off(dataChannel, onData);
-		ipc.off(errorChannel, onError);
+		// ipc.off(dataChannel, onData);
+		// ipc.off(errorChannel, onError);
	};

	const onData = (event, result) => {
		cleanup();
		resolve(result);
	};

	const onError = (event, error) => {
		cleanup();
		reject(deserializeError(error));
	};

	ipc.once(dataChannel, onData);
	ipc.once(errorChannel, onError);
```
In my opinion the `off` calls are redundant anyways, because the listeners are attached using `once`, so they will be removed automatically after first event regardless (and that's what happens after my change, as far as I know). I feel like the double removal might be the issue here.

**Comment by sindresorhus** [comment]
> In my opinion the off calls are redundant anyways,

No. Without the cleanup, if data is called, the error one is never cleaned up.

**Comment by maciejewiczow** [comment]
Okay, I haven't thought of this. But nonetheless, something strange is going on here. After another bit of poking around I discovered that adding any listener to ipc emitter before first call to `callMain` also fixes this issue. I have even less of an idea now as to what might be the cause of this behavior.

**Comment by danwt** [comment]
I have the same issue

**Comment by maciejewiczow** [comment]
For now, my temporary workaround around the issue is to put the following code in the root file of the part of the project that uses `electron-better-ipc` 
```ts
if (ipcMain !== undefined)
    ipcMain.addListener('fix-event-798e09ad-0ec6-5877-a214-d552934468ff', () => {});

if (ipcRenderer !== undefined)
    ipcRenderer.addListener('fix-event-79558e00-29ef-5c7f-84bd-0bcd9a0c5cf3', () => {});
```
No idea why this changes anything though, just stumbled upon it randomly

**Comment by pixelart7** [comment]
I am having similar issue, and I can confirm that the workaround is working.

**Comment by mferris77** [comment]
Also confirming the workaround solves this issue. This is the code at the top of my file where I invoke electron-better-ipc:

```
const { app, remote, ipcMain, ipcRenderer } = require("electron");
const { ipcRenderer: ipc } = require("electron-better-ipc");

// fix for electron-better-ipc
// from https://github.com/sindresorhus/electron-better-ipc/issues/35
if (ipcMain !== undefined)
  ipcMain.addListener(
    "fix-event-798e09ad-0ec6-5877-a214-d552934468ff",
    () => {}
  );

if (ipcRenderer !== undefined)
  ipcRenderer.addListener(
    "fix-event-79558e00-29ef-5c7f-84bd-0bcd9a0c5cf3",
    () => {}
  );
//end fix

```

**Comment by banben** [comment]
Have the same problem too.

**Comment by tjklemz** [comment]
Same. Makes the library unusable. :(

**Comment by cannc4** [comment]
I can confirm that the workaround works. Odd solution though :)

**Comment by macharborguy** [comment]
Confirmed both the bug as well as the workaround being successful.

For those who cannot get the workaround, well, working, make sure that the **addListener** method is chained after the variable containing the specific IPC.

so if you alias your ipcMain or ipcRenderer methods and rename them to simply 'ipc', you will need to add the listeners to 'ipc'

**Comment by user896724** [comment]
Note that you need the applicable part of code in both environments - renderer and main.  I have the following:

In src/main.js (main):

```
let {
	app,
	// etc
	ipcMain,
} = require("electron");

let {ipcMain: ipc} = require("electron-better-ipc");

// HACK for https://github.com/sindresorhus/electron-better-ipc/issues/35
ipcMain.addListener("fix-event-798e09ad-0ec6-5877-a214-d552934468ff", () => {});
```

In src/app.js (renderer):

```
// HACK for https://github.com/sindresorhus/electron-better-ipc/issues/35
require("electron").ipcRenderer.addListener("fix-event-79558e00-29ef-5c7f-84bd-0bcd9a0c5cf3", () => {});
```
